File: 3451-string-compression-iii/3451-string-compression-iii.py
class Solution:
    def compressedString(self, word: str) -> str:
        compressed = []
        i = 0
        while i <len(word):
            count = 1
            while i+count <len(word) and word[i]== word[i+count] and count<9:
                count+=1
            compressed.append(f'{count}{word[i]}')
            i+=count
        return ''.join(compressed)
        

File: 3451-string-compression-iii/3451-string-compression-iii.cpp
class Solution {
public:
    string compressedString(string word) {
        string s = "";
        int i = 0, n = (int)word.length();
        stack<char> stk;
        while (i < n) {
            stk.push(word[i]);
            int j = i + 1;
            while (j < n && stk.top() == word[j] && (int)stk.size() < 9) stk.push(word[j]), ++j;
            s.push_back('0' + (int)stk.size());
            s.push_back(stk.top());
            while (!stk.empty()) stk.pop();
            i = j;
        }
        return s;
    }
};